spec/factories/course_survey_question_options.rb

```diff
@@ -11,7 +11,11 @@
     weight { last_weight ? last_weight + 1 : 0 }
 
     trait :with_attachment do
-      attachment_reference
+      after(:build) do |question_option|
+        attachment = create(:attachment_reference)
+        attachment.save!
```

Comment:
Do we need this line? I think `create` also does persistence, if I am not mistaken.

Reply:
Good catch! thanks and updated